src/model/formats/foreflight-utils.ts

```diff
@@ -22,6 +28,34 @@ export class ForeFlightUtils {
   static readonly encoder = new TextEncoder();
   static readonly decoder = new TextDecoder();
 
+  public static readonly CHECKLIST_ITEM_PREFIXES = new Map([
+    [ChecklistItem_Type.ITEM_PLAINTEXT, ''],
+    [ChecklistItem_Type.ITEM_NOTE, 'NOTE: '],
+    [ChecklistItem_Type.ITEM_CAUTION, 'CAUTION: '],
+    [ChecklistItem_Type.ITEM_WARNING, 'WARNING: '],
+  ]);
+
+  public static readonly CHECKLIST_ITEM_TYPES = new Map(
+    [...ForeFlightUtils.CHECKLIST_ITEM_PREFIXES.entries()].map(
+      (item) => item.reverse() as [string, ChecklistItem_Type],
```

Comment:
Minor: The "as" here should not be needed if you do `new Map<ChecklistItem_Type, string>` and `new Map<string, ChecklistItem_Type>` above.
(Typescript tries to be strongly typed, so avoiding casting like this is a good idea)

Reply:
Thanks for the nudge, I didn't like the explicit cast either, but TypeScript's inference engine is just too stupid. The problem is that it couldn't infer that `reverse`ing a tuple would produce a tuple with swapped elements. So your suggestion would unfortunately not work.

However, it seems I found a solution in a2633eb - I implemented a Haskell-style `Data.Tuple` generic `swap` function, and now TypeScript is able to infer the types correctly. This has to do with the fact that tuples are not really tuples at all, but rather arrays, sigh.